packages/bff/src/model/api/converters/catalogClientApiConverter.ts

```diff
@@ -0,0 +1,214 @@
+/* eslint-disable functional/immutable-data */
+/* eslint-disable max-params */
+import { DescriptorWithOnlyAttributes } from "pagopa-interop-agreement-lifecycle";
+import {
+  agreementApi,
+  bffApi,
+  catalogApi,
+  tenantApi,
+  attributeRegistryApi,
+} from "pagopa-interop-api-clients";
+import {
+  EServiceAttribute,
+  tenantMailKind,
+  unsafeBrandId,
+} from "pagopa-interop-models";
+import { attributeNotExists } from "../../domain/errors.js";
+import { agreementApiState } from "../agreementTypes.js";
+import { descriptorApiState } from "../catalogTypes.js";
+
+export function toEserviceCatalogProcessQueryParams(
+  queryParams: bffApi.BffGetCatalogQueryParam
+): catalogApi.GetCatalogQueryParam {
+  return {
+    ...queryParams,
+    eservicesIds: [],
+    name: queryParams.q,
+  };
+}
+
+export function toBffCatalogApiEService(
+  eservice: catalogApi.EService,
+  producerTenant: tenantApi.Tenant,
+  hasCertifiedAttributes: boolean,
+  isRequesterEqProducer: boolean,
+  activeDescriptor?: catalogApi.EServiceDescriptor,
+  agreement?: agreementApi.Agreement
+): bffApi.CatalogEService {
+  const isUpgradable = (agreement: agreementApi.Agreement): boolean => {
+    const eserviceDescriptor = eservice.descriptors.find(
+      (e) => e.id === agreement.descriptorId
+    );
+
+    return (
+      eserviceDescriptor !== undefined &&
+      eservice.descriptors
+        .filter((d) => Number(d.version) > Number(eserviceDescriptor.version))
+        .find(
+          (d) =>
+            (d.state === descriptorApiState.PUBLISHED ||
+              d.state === descriptorApiState.SUSPENDED) &&
+            (agreement.state === agreementApiState.ACTIVE ||
+              agreement.state === agreementApiState.SUSPENDED)
+        ) !== undefined
+    );
+  };
```

Comment:
For reference: an almost the same function is present [here](https://github.com/pagopa/interop-be-monorepo/pull/677/files#diff-2390fbcbfb4d04605e4a7fbf90e48501f57966298e6a34b21745523f071cb129R177-R188) in another PR

Reply:
@Carminepo2 in this commit have introduced a new file `mappers.ts`, my intent is to move here the commons function used to pick, transform or evaluate data, not for Validation and API's convert function help function. 
Also in this next [PR](https://github.com/pagopa/interop-be-monorepo/pull/766/files#diff-cfe8c1e4cc20ca72b08043360620e945571633fd49bbe7302730aa5d6a61a1a2R30) I'm using this approach, but in  this case probably I prefer to place this method in mapper.

Reply:
this [commit](https://github.com/pagopa/interop-be-monorepo/pull/712/commits/7dc73a20f75f65a16e938f00100ea8da9b22919f) move this logic in a specific file